go-controller/pkg/util/net_linux.go

```diff
@@ -349,25 +349,13 @@ func LinkRoutesAdd(link netlink.Link, gwIP net.IP, subnets []*net.IPNet, mtu int
 	return nil
 }
 
-// IsAnyNetwork checks if the argument network is an any network for ipv4 or ipv6.
-func IsAnyNetwork(ipNet *net.IPNet) bool {
+// IsNilOrAnyNetwork checks if the argument network is nil or an any network for ipv4 or ipv6.
+func IsNilOrAnyNetwork(ipNet *net.IPNet) bool {
 	if ipNet == nil {
-		return false
-	}
-	ones, _ := ipNet.Mask.Size()
-	//v4
-	if ipNet.IP.To4() != nil {
-		v4Any := net.ParseIP("0.0.0.0")
-		if ipNet.IP.Equal(v4Any) && ones == 0 {
-			return true
-		}
-	} else {
-		v6Any := net.ParseIP("::")
-		if ipNet.IP.Equal(v6Any) && ones == 0 {
-			return true
-		}
+		return true
```

Comment:
For upstream, it's unlikely to have a regression on the `route.Dst`'s return value for ANY(e.g. netlink got downgrade or revert to previous behavior).

 This change makes this API backward compatible with old netlink and previous behavior -- nil is treated as ANY.

Reply:
Hey! Thanks for the change. May i ask why you want it backward compat ? Are you consuming ovn-k and replacing the netlink version?

Reply:
or are you just try to prevent regression if someone cherry-picked this commit and didnt include the netlink lib bump?